joblib/test/test_numpy_pickle.py

```diff
@@ -156,21 +153,35 @@ def test_numpy_persistence():
     filename = env['filename']
     rnd = np.random.RandomState(0)
     a = rnd.random_sample((10, 2))
-    for compress, cache_size in ((0, 0), (1, 0), (1, 10)):
+    for compress, cache_size in ((0, None), (0, None),
+                                 (0, 0), (1, 0),
+                                 (0, 10), (1, 10)):
         # We use 'a.T' to have a non C-contiguous array.
         for index, obj in enumerate(((a,), (a.T,), (a, a), [a, a, a])):
             # Change the file name to avoid side effects between tests
             this_filename = filename + str(random.randint(0, 1000))
-            filenames = numpy_pickle.dump(obj, this_filename,
-                                          compress=compress,
-                                          cache_size=cache_size)
+
+            # Check warning if cache size is set
+            with warnings.catch_warnings(record=True) as w:
+                warnings.simplefilter("always")
+                filenames = numpy_pickle.dump(obj, this_filename,
+                                              compress=compress,
+                                              cache_size=cache_size)
+                nose.tools.assert_equal(len(w),
+                                        1 if cache_size is not None else 0)
+                for warn in w:
+                    nose.tools.assert_equal(warn.category, DeprecationWarning)
+                    nose.tools.assert_equal(warn.message.args[0],
+                                            'Cache size is deprecated and '
+                                            'will be ignored.')
+
             # Check that one file was created per array
-            if not compress:
-                nose.tools.assert_equal(len(filenames), len(obj) + 1)
-            # Check that these files do exist
-            for file in filenames:
+            for fname in filenames:
+                if not compress:
+                    nose.tools.assert_equal(fname, this_filename)
+                # Check that these files do exist
```

Comment:
There is only one file now: "this file does exist" rather than "these files do exist"